src/terminal/parser/stateMachine.hpp

```diff
@@ -24,7 +24,7 @@ namespace Microsoft::Console::VirtualTerminal
     // for parameter values. 65535 is what XTerm and VTE support.
     // GH#12977: We must use 65535 to properly parse win32-input-mode
     // sequences, which transmit the UTF-16 character value as a parameter.
-    constexpr VTInt MAX_PARAMETER_VALUE = 65535;
+    constexpr VTInt MAX_PARAMETER_VALUE = 0xffffff;

```

Comment:
With this larger range, I think we would now need to add a clamp to the `REP` escape sequence. 65535 should be fine, since that's what most other terminals support, but 16M is a potential DOS attack.

You should probably also update this comment to reflect the new range and why it is what it is.

Reply:
Wow, I just noticed that I confused #15390 with rect fill operations. I'm not sure why I made this connection in the first place... It's completely wrong. I'll back out the Unicode change and just leave the new primitive in there. They're still useful in a way, since they remove the `IsGlyphFullWidth` call.

In any case, I know what the `REP` sequence is, but when I search for that in the project I can't seem to find. Do you happen to know where it's implemented?

Reply:
It's kind of a special case, because it's implemented in the `OutputStateMachine`.

https://github.com/microsoft/terminal/blob/612b00cd445470fa85d2f71dc94930f03ebe4823/src/terminal/parser/OutputStateMachineEngine.cpp#L564-L577

Reply:
And I completely forgot about #15390. As things are currently implemented, I think that's probably going to be a pain to fix. But if you're doing a rewrite of the state machine code, it might be worth considering using UTF-32 as the base character, because in that case the `REP` would just work as is.